### PR TITLE
feat: Introduce upload validation to prevent XSS exploits with SVG

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

This PR introduces upload validation hooks to prevent users from intentionally or unintentionally upload malicious SVG or HTML files.

The default setting is to prevent all HTML file uploads and to check for potential scripts in SVG files.

To keep things simple, the SVG file check looks for certain keywords or obfuscation tactics. IMHO, it will lean towards false positives (i.e. rejecting files that in fact are OK).

The default validator can be replaced by a more sophisticated custom validator. However, since bleach is sunset, it is a non-trivial task to create one.

## Related resources

* #1360 adds `rel="noopener noreferrer"`
* #...

## Checklist

* [x] I have opened this pull request against ``master``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-filer/blob/master/docs/development.rst#contributing) and I have joined #workgroup-pr-review on 
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: docs/validation.rst [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        ...
+        if validation_failed:
+            # Validation error will be forwarded to the user as admin message
+            raise FileValidationError(
+                _('File "{}"': Upload rejected since file contains JavaScript code").format(file_name)
+            )
+
+
+Build-in validators
+-------------------
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply docs/validation.rst:76]: Not sure. But I counted twice: The number of dashes is correct.

File: docs/validation.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    If you have legitimate SVG files that contain either ``<script`` or
+    ``javascript:`` as byte sequences try escaping the ``<`` and ``:``.
+
+Clearly, the validator can be improved by parsing the SVG's xml code, but
+this could be error-prone and we decided to go with the potentially too strict
+but simpler method.
+
+Common validator settings
+-------------------------
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply docs/validation.rst:119]: I'm not sure why...

File: filer/templates/admin/filer/tools/detail_info.html [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
                 {% if original.file.exists %}
-                    <a href="{{ original.url }}" target="_blank" class="btn" download="{{ original.original_filename }}">
-                        <span class="icon fa fa-download"></span>
-                        &nbsp;<span>{% trans "Download" %}</span>
+                    <a href="{{ original.url }}"
+                       rel="noreferrer noopener"
+                       target="_blank" class="button"
+                       download="{{ original.original_filename }}">
+                        <span class="icon fa fa-download filer-icon filer-icon-download"></span>&nbsp;
+                        <span>{% trans "Download" %}</span>
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply filer/templates/admin/filer/tools/detail_info.html]: Done :-)

**Comment by fsbraun** [comment]
@marksweb I just implemented the requested changes and added another setting: `FILER_MIME_TYPE_WHITELIST` which would allow to whitelist accepted mime types, effectively barring all others. So, one could restrict filer to only accept images and no other formats (including all thinkable executable files)

**Comment by fsbraun** [comment]
@jrief pointed me to reportlab's ability to rewrite SVG files, thereby eliminating script tags and event attributes. I added this as an experimental feature.

**Review by vinitkumar — APPROVED** [review]
Looks great, Let's get it merged and released.

**Review by marksweb — COMMENTED** [review]
Getting through this slowly.

**Pull request merged**